src/Features/Core/Portable/InheritanceMargin/AbstractInheritanceMarginService_Helpers.cs

```diff
@@ -96,6 +96,15 @@ internal abstract partial class AbstractInheritanceMarginService
                 if (member == null || !CanHaveInheritanceTarget(member))
                     continue;
 
+                // Make sure the identifier declaration's span is within the spanToSearch.
+                // This is important because for example, for a class with big body, when the tagger is asking for the tag within the body of the class,
+                // we don't want to return the tag of the class identifier.
+                var declarationToken = GetDeclarationToken(memberDeclarationNode);
+                if (!spanToSearch.Contains(declarationToken.Span))
```

Comment:
`spanToSearch` comes from https://github.com/dotnet/roslyn/blob/0447e0c1549f0f9a2706853172f00507b352773f/src/VisualStudio/Core/Def/InheritanceMargin/InheritanceMarginTaggerProvider.cs#L91C28-L91C37

So it is assuming the tagger infra to always ask the correct region to tag.
(Which I believe it should)

Reply:
can you move this above the semantic checks above.  this syntax check is much faster than the symbol checks.